pkg/webhooks/provisioning/v1alpha1/defaulter.go

```diff
@@ -0,0 +1,67 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +kubebuilder:webhook:path=/mutate-provisioning-karpenter-sh-v1alpha1-provisioner,mutating=true,sideEffects=None,failurePolicy=fail,groups=provisioning.karpenter.sh,resources=provisioners,verbs=create;update,versions=v1alpha1,name=mutation.provisioning.karpenter.sh
+
+package v1alpha1
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	provisioningv1alpha1 "github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/awslabs/karpenter/pkg/utils/functional"
+	"knative.dev/pkg/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// podDefaulter validates Pods
+type Defaulter struct {
+	decoder       *admission.Decoder
+}
+
+// Path of the webhook handler
+func (v *Defaulter) Path() string {
+	return "/mutate-provisioning-karpenter-sh-v1alpha1-provisioner"
+}
+
+// InjectDecoder injects the decoder.
+func (v *Defaulter) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
+
+// podDefaulter admits a pod if a specific annotation exists.
+func (v *Defaulter) Handle(ctx context.Context, req admission.Request) admission.Response {
+	provisioner := &provisioningv1alpha1.Provisioner{}
+	err := v.decoder.Decode(req, provisioner)
+	if err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+	functional.ExecuteAll(
```

Comment:
I'm a bit confused by these lines - is `ExecuteAll` needed here?

Reply:
I was just trying to build a coherent structure around running a bunch of defaulting functions. I can rip this out though.